docs/external-plugin-messaging.md

```diff
@@ -0,0 +1,87 @@
+# External Plugin Messaging
+
+Other Plugin Hub plugins can publish a `PluginMessage` via `EventBus#post` that instructs Dink to submit a webhook request.
+
+Users can opt-out of this capability by disabling `External Plugin Requests > Enable External Plugin Notifications`.
+
+Plugins can request that a screenshot is included with the notification, but users can also opt-out by
+disabling `External Plugin Requests > Send Image` (default: on) and `External Plugin Requests > Allow Overriding 'Send Image'` (default: off).
+
+Plugins can include a Discord url for the webhook, otherwise Dink will utilize `External Webhook Override`
+(or `Primary Webhook URLs` if an external url override is not specified).
+If a plugin requests a non-Discord url, it will be ignored in favor of the Dink configuration.
+
+Otherwise, external plugins have a great deal of customization for the webhook body.
+
+## Payload
+
+The `namespace` for the `PluginMessage` should be `dink` and the `name` should be `notify`.
+
+The `Map<String, Object>` that is supplied to `PluginMessage` will be converted into [`ExternalNotificationRequest`](../src/main/java/dinkplugin/domain/ExternalNotificationRequest.java).
+
+| Field            | Required | Type             | Description                                                                                                                                                                             |
+| ---------------- | -------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `text`           | Y        | String           | The body text of the notification. This field supports templating (see `replacements` below) and by default `%USERNAME%` is an available replacement.                                   |
+| `url`            | N        | String           | The Discord URLs that the notification should be sent to (newline separated).                                                                                                           |
```

Comment:
should this be `urls` instead since it can accept multiple URLs? only fear would be that it implies a list of some kind instead

Reply:
Yeah had a similar fear + majority of use cases would only specify a single url

We could also convert the field to a list of strings, but that'd be unergonomic for most folks that only have 1 url